packages/react/src/user-preferences/userPreferences.ts

```diff
@@ -0,0 +1,51 @@
+export type UserPreferencesData = {
+  theme?: string;
+  lang?: string;
+};
+
+export type PreferencesUpdateCallback = (data?: UserPreferencesData) => void;
+
+export class UserPreferences {
+  #preferences?: UserPreferencesData;
+  #listeners: Set<PreferencesUpdateCallback> = new Set();
+
+  constructor() {
+    this.#setupMessageListener();
+    this.#init();
+  }
+
+  #setupMessageListener() {
+    window.addEventListener('message', (event) => {
+      const { type, payload } = event.data;
+
+      switch (type) {
+        case '@microapp:userpreferences':
+          this.#preferences = payload;
+          this.#notifyListeners();
+          break;
+
+        default:
+          break;
+      }
+    });
+  }
+
+  #init() {
+    // TODO: Change targetOrigin from "*" to the host origin for security reasons
```

Comment:
As this is the `Production` strategy, you can create a const like `PRODUCTION_MARKETPLACE_HOST_URL` in some shared file of the `Runtime` and use it here.